pandas/core/reshape/merge.py

```diff
@@ -1593,6 +1594,15 @@ def _factorize_keys(lk, rk, sort=True):
         lk = lk.values
         rk = rk.values
 
+    elif (is_extension_array_dtype(lk.dtype) and
```

Comment:
maybe should do this after the categorical check?

Reply:
The categorical check is already in the main `if` block.

I'm going to move that up to the same if / elif as the datetimetz, and the new extension dtype check.